### ISSUE TITLE
cljdocify cljdoc-analyzer

### ISSUE THREAD

**Issue by lread** [lread/cljdoc-analyzer#11]
Add appropriate cljdocisms to cljdoc-analyzer itself
* `^:no-doc` metadata
* `doc/cljdoc.edn`

**Comment by lread** [comment]
Ironicaly, cljdoc-analyzer cannot analyze itself as it tries to parse metagetta source. We could add an option to exclude-dirs from analysis.

**Comment by martinklepsch** [comment]
What you could do is also packaging metagetta as it’s own jar and then in the cljdoc analyzer project use something like MrRanderson to online the sources. This would also remove the namespace conflicts. I’m not 100% sure if that’s the issue but might be worth a look. 

Might also make it easier for people to use metagetta on its own.

**Comment by lread** [comment]
Thanks @martinklepsch, I guess I am weighing the technical awkwardness of including metagetta as a subproject against the maintenance awkwardness of splitting it out.

The MrAnderson idea is interesting and merits its own git issue. I’m wondering it we used MrAnderson throughout we would even need to treat metagetta as a subproject and subprocess.

**Comment by martinklepsch** [comment]
That would mean people could not use metagetta on its own which I think might be desirable. It’s essentially a codox Alternative that’s more focused on programmatic data extraction. 

@borkdude is doing a great job naming and announcing all the projects in the clj-kondo periphery so maybe it would make sense for us to follow a similar approach and just announce it as a separate thing. As changes won’t be too frequent the burden of packaging it as a separate library shouldn’t be too high (just use Leiningen).

And then the cljdoc analyzer project —  is much more specific essentially trying to create more isolation and construct the class path in more specific ways.

I dont know - what do you think?

**Comment by lread** [comment]
Thanks @martinklepsch.

Because metagetta focuses on the use case of working from jars, rather than source trees, it is not currently really a codox alternative (it could become that, but we decided not to take on that cost initially). Because of its focus on the jar, I didn't really see it as being useful to others outside of cljdoc-analyzer. But yeah, I could be totally wrong, I guess feedback from the community will tell us. 

I currently see cljdoc-analyzer as codox data extraction - but for jars.  I  am actually using it as that in my personally little [diff-apis project](https://github.com/lread/diff-apis).

I also did not necessarily want to encourage the support burden of a separate metagetta project and/or jar, at least initially.

There is nothing stopping us from extracting out metagetta to its own project in the future, but right now, I am thinking I'll keep it embedded.  I'm currently working on packing it as a jar which I'll bundle inside the cljoc analyzer jar.  

Anyway, I very much appreciate your thoughts on this topic!  Continued discussion/objections/ideas are most welcome!
